### PR TITLE
TypeScript improvements

### LINKED ISSUE #69: Allow spaced-comment for TypeScript type imports
TypeScript has this special syntax:

```js
/// <reference types="jest" />
```

We should configure `spaced-comment` to allow this:

```
    'spaced-comment': [
      'error',
      'always',
      {
        line: {
          markers: ['/'],
          exceptions: ['/']
        }
      }
    ]
```

### LINKED ISSUE #70: Disable `import/export` for TypeScript
This creates false positives when there are overloads.

E.g.:
```js
import useQueryParamString from './useQueryParamString';

export default function useQueryParamNumber(
  name: string,
  required: true
): number;
export default function useQueryParamNumber(
  name: string,
  required?: false
): number | undefined;
export default function useQueryParamNumber(name: string, required?: any) {
  const value = useQueryParamString(name, required);
  return value ? parseFloat(value) : undefined;
}
```

### LINKED ISSUE #78: Disable `@typescript-eslint/no-empty-function`

### LINKED ISSUE #83: Disable @typescript-eslint/no-empty-interface
Here's a valid use case with the library theme: https://emotion.sh/docs/typescript#define-a-theme

### LINKED ISSUE #85: describe.only is not being detected
Currently this is being used: https://github.com/molindo/eslint-config-molindo/blob/master/cypress.js

This could be used: https://www.npmjs.com/package/eslint-plugin-no-only-tests

### PR DESCRIPTION
**Pull request by amannn**
Fixes #69
Fixes #70
Fixes #78
Fixes #83
Fixes #85

Pre-release:

```
eslint-config-molindo@6.0.0-alpha.6
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "eslint-plugin-sort-destructure-keys": "^1.3.5",
     "eslint-plugin-unicorn": "^20.1.0",
     "prettier": "^2.0.0",
-    "typescript": "^3.0.0"
+    "typescript": "^4.0.0"
```
Comment (amannn) [inline package.json:50]: Maybe we should also update `@typescript-eslint/parser` and `@typescript-eslint/eslint-plugin`

**Review by symn — APPROVED** [review]
lgtm

File: package.json [hunk 2]
```diff
@@ -25,29 +25,28 @@
...
   },
   "peerDependencies": {
-    "eslint": "^7.0.0"
+    "eslint": "^8.0.0"
   },
   "dependencies": {
-    "@rushstack/eslint-patch": "^1.0.2",
-    "@typescript-eslint/eslint-plugin": "^3.6.0",
-    "@typescript-eslint/parser": "^3.0.0",
-    "babel-eslint": "^10.0.0",
```
Comment (amannn) [inline package.json:38]: Seems like we no longer need this parser.

File: .eslintrc.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
-module.exports = Object.assign({}, require('./javascript.js'), {
-  env: {
-    node: true,
-    jest: true
+module.exports = Object.assign(
+  {},
+  {
+    extends: ['./javascript', './typescript', './cypress', './jest', './react'],
```
Comment (amannn) [inline .eslintrc.js:4]: Load all rule definitions to see if they are valid.

**Review by symn — APPROVED** [review]
lgtm

**Pull request merged**